### ISSUE TITLE
Adding duplicate messages in AddMessage() is not handled properly

### ISSUE THREAD

**Issue by yzzyx** [zenhack/go.notmuch#27]
In db.go, method `AddMessage`, the check for errors does not handle duplicate messages the same way that the notmuch-library does.

A quick look in `lib/add-message.cc` in the notmuch source-code 
https://git.notmuchmail.org/git?p=notmuch;a=blob;f=lib/add-message.cc;h=485debad225239211e9a6f165f24be519080a0ed;hb=HEAD#l574 shows that the message information is set on success _or_ if a duplicate message has been found.

The relevant section being:
```
    if (message) {
	if ((ret == NOTMUCH_STATUS_SUCCESS ||
	     ret == NOTMUCH_STATUS_DUPLICATE_MESSAGE_ID) && message_ret)
	    *message_ret = message;
	else
	    notmuch_message_destroy (message);
    }
```

We therefore have to update the AddMessage-function to both make sure that the gc cleans our message up, and also make it available to the caller.

**Comment by brunnre8** [comment]
```
 * If 'message' is not NULL, then, on successful return
 * (NOTMUCH_STATUS_SUCCESS or NOTMUCH_STATUS_DUPLICATE_MESSAGE_ID) '*message'
 * will be initialized to a message object that can be used for things
 * such as adding tags to the just-added message. The user should call
 * notmuch_message_destroy when done with the message. On any failure
 * '*message' will be set to NULL.
```
From the header, says the same thing in actual words :wink:

**Comment by zenhack** [comment]
Fixed by #28, closing.
